### PR TITLE
Rooted product of graphs

### PR DESCRIPTION
**Pull request by mcognetta**
Addresses issue https://github.com/networkx/networkx/issues/2017

Implements code to construct the rooted product of two graphs. Relevant information can be found at https://en.wikipedia.org/wiki/Rooted_product_of_graphs

This is my first pull request to this product so please let me know if there is anything I should fix.

### CODE REVIEW

File: networkx/algorithms/operators/product.py [hunk 1]
```diff
@@ -432,3 +434,41 @@ def power(G, k):
...
+    A new graph is constructed representing the rooted product of
+    the inputted graphs, G and H, with a root in H.
+
+    The nodes of R are the Cartesian Product of the nodes of G and H. The
+    nodes of G and H are not relabeled.
+
+    """
+
+    if G.is_multigraph() or H.is_multigraph():
+        raise nx.NetworkXError('G and H must not be multigraphs')
```
Comment (dschult) [inline networkx/algorithms/operators/product.py]: You should use the "decorator" called ```not_implemented_for```.   see line 352 with the function ```product```

Can you add a test or two in tests/test_product.py?

Thanks!

**Comment by mcognetta** [comment]
Yes, sorry I forgot to track my tests. I will do this tonight. 

Thanks.

File: networkx/algorithms/operators/product.py [hunk 2]
```diff
@@ -432,3 +434,41 @@ def power(G, k):
...
+
+    if root not in H:
+        raise nx.NetworkXError('root must be a vertex in H')
+
+    R = nx.Graph()
+    R.add_nodes_from(product(G,H))
+    
+    R.add_edges_from([((e[0],root),(e[1],root)) for e in G.edges()]
```
Comment (jfinkels) [inline networkx/algorithms/operators/product.py]: This is probably clearer as the following two lines:

    R.add_edges_from(((u, root), (v, root)) for u, v in G.edges())
    R.add_edges_from(((u, w), (v, w)) for w in G for u, v in H.edges())

Note the lack of square brackets: this is a generator expression, so we don't need to create the full list in memory, which may be very expensive in terms of space for large graphs.

**Comment by mcognetta** [comment]
Can you also restart tests on this pull request? Thanks.

**Pull request closed without merging**